### PR TITLE
fix(#1975): remove console warning for deprecated props

### PR DESCRIPTION
**Pull request by syedszeeshan**
# Before (the change)
console warning for aria*
image

console warning for variant="nocolor"
image

# After (the change)
No such console warnings anymore.

## Make sure that you've checked the boxes below before you submit the PR

- [x] I have read and followed the [setup steps](https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3086385489/Contribution+-+web+content#%5BinlineExtension%5DCode-contributions)
- [ ] I have created necessary unit tests
- [x] I have tested the functionality in both React and Angular.

## Steps needed to test

Add the code below in Angular playground and toggle the commented line to verify that the `variant="dark"` is the correct replacement for the `variant="nocolor"`
```
<!-- <goa-icon-button icon="close" size="small" variant="nocolor"></goa-icon-button> -->
<goa-icon-button icon="close" size="small" variant="dark"></goa-icon-button>
<goa-icon-button icon="close" size="small" variant="color"></goa-icon-button>
```

### CODE REVIEW

**Comment by syedszeeshan** [comment]
Discussed with Chris and Vanessa and confirmed that we need these properties, just suppress the console warning for ariacontrols and ariaexpanded.

The closest tothe  `varaint="nocolor"` is `variant="dark"`, sample code is in the description.

**Comment by lizhuomeng71** [comment]
I still got the console error for  variant="nocolor" for both react and angular

image

**Comment by syedszeeshan** [comment]
@lizhuomeng71 
The variant="nocolor" option is deprecated but still available.... however if devs use it (like you did in your example) it will give the console warning. What I have done in this PR is to remove all such occurences in "our codebase" and replace it with variant="dark", but there is nothing stopping you from using variant="nocolor".

**Comment by lizhuomeng71** [comment]
issue is verified on react and angular,

File: libs/web-components/src/components/modal/Modal.svelte [hunk 1]
```diff
@@ -188,7 +191,7 @@
         out:fly={{ delay: _transitionTime, duration: _transitionTime, y: -100 }}
         class="modal-pane"
         tabindex="-1"
-        role={role}
+        {role}
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/modal/Modal.svelte:194]: I'm pretty sure the `role` property requires it to be `role={role}`

**Review by ArakTaiRoth — APPROVED** [review]

**Pull request merged**